###############################################################################
#
# Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.
#
# @(#)pciids-version.patch	1.1	08/10/13

Use the snapshot date in the pci.ids file to generate a SCCS-like
version info tag, so that running 'what pcidata.so' or 'what scanpci.so'
will show what version of pci.ids data is included.

--- hw/xfree86/scanpci/pciid2c.pl	Wed Oct  4 14:46:03 2006
+++ hw/xfree86/scanpci/pciid2c.pl	Thu Oct  9 18:11:53 2008
@@ -110,6 +110,8 @@
 	}
     } elsif (/^\t\tC\s+([0-9a-f]{4})/) {
 	$classes{$vendor}{$device} = $1;
+    } elsif (/^#\s+(.* snapshot .*)$/) {
+        $snapshot_version = $1;
     }
     # Ignore all other lines.
 }
@@ -162,6 +164,14 @@
 
 ";
 
+print qq(
+\#ifdef __SUNPRO_C
+\#pragma ident "\@(\#)pci.ids\t$snapshot_version"
+\#else
+const char sccsid[] = "\@(\#)pci.ids\t$snapshot_version";
+\#endif
+);
+
 # The following #ifdefs are used:
 #  - INIT_SUBSYS_INFO -- initialise subsystem data
 #  - INIT_VENDOR_SUBSYS_INFO -- initialise a vendor<->subsystem table.
